circuits/cpp/barretenberg/cpp/src/barretenberg/stdlib/recursion/honk/transcript/trancript.hpp

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include "barretenberg/ecc/curves/bn254/fq.hpp"
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/ecc/curves/bn254/g1.hpp"
+#include "barretenberg/honk/sumcheck/polynomials/univariate.hpp"
+#include "barretenberg/honk/transcript/transcript.hpp"
+
+#include "barretenberg/stdlib/primitives/bigfield/bigfield.hpp"
+#include "barretenberg/stdlib/primitives/biggroup/biggroup.hpp"
+#include "barretenberg/stdlib/primitives/field/field.hpp"
+
+//TODO(luke): this namespace will be sensible once stdlib is moved out of the plonk namespace
+namespace proof_system::plonk::stdlib::recursion::honk {
+template <typename Builder> class Transcript {
+  public:
+    using field_pt = field_t<Builder>;
+    using witness_pt = witness_t<Builder>;
+    using fq_pt = bigfield<Builder, barretenberg::Bn254FqParams>;
+    using group_pt = element<Builder, fq_pt, field_pt, barretenberg::g1>;
+    using FF = barretenberg::fr;
+    using Commitment = barretenberg::g1::affine_element;
+    using VerifierTranscript = proof_system::honk::VerifierTranscript<FF>;
+    template <size_t LENGTH> using Univariate = proof_system::honk::sumcheck::Univariate<FF, LENGTH>;
+
+    static constexpr size_t HASH_OUTPUT_SIZE = 32; // WORKTODO: Duplicated from native transcript
+
+    VerifierTranscript native_transcript;
+    Builder* builder;
+
+    Transcript(Builder* builder, auto proof_data)
+        : native_transcript(proof_data)
+        , builder(builder){};
+
+    /**
+     * @brief Get the underlying native transcript manifest (primarily for debugging)
+     *
+     */
+    auto get_manifest() const { return native_transcript.get_manifest(); };
+
+    /**
+     * @brief Compute the challenges (more than 1) indicated by labels
+     *
+     * @tparam Strings
+     * @param labels Names of the challenges to be computed
+     * @return std::array<FF, sizeof...(Strings)> Array of challenges
+     */
+    template <typename... Strings> std::array<field_pt, sizeof...(Strings)> get_challenges(const Strings&... labels)
+    {
+        // Compute the indicated challenges from the native transcript
+        constexpr size_t num_challenges = sizeof...(Strings);
+        std::array<FF, num_challenges> native_challenges{};
+        native_challenges = native_transcript.get_challenges(labels...);
+
+        /*
+         * TODO(luke): Do stdlib hashing here. E.g., for the current pedersen/blake setup, we could write data into a
```

Comment:
Could you make an issue int he aztec-packages repo and put in the appropriate slot in the project board and link here?

Reply:
done, #1351